code.test.js

```diff
@@ -0,0 +1,49 @@
+const fs = require('fs');
+const assert = require('assert');
+
+// Load the code.js file which contains the dijkstra function
+eval(fs.readFileSync('code.js') + '');
+
+// === Test 1: Single node ===
+let graph = [[]]; // No edges
+let result = dijkstra(graph, 0);
+assert.deepStrictEqual(result, [0]);
+
+// === Test 2: Two connected nodes ===
+graph = [
```

Comment:
This doesn't look like a standard graph representation -- can you use an adjacency matrix or adjacency list please?